spheres.js

```diff
@@ -238,7 +238,26 @@ function IntersectRaySphere(origin, direction, sphere)
 
 	// minor optimization
 	// avoid recomputing
-	const k1Times2 = k1 << 1;
+	// premature optimization is not the root of all evil
+	// being cute is the root of all evil....
+	// and lest we forget, k1 IS NOT AN INTEGER
+	// ...the bit shifting "doesn't work"
+	// and of course there were no unit tests for this.
+	// might not have caught it either since k1 is the dot product
+	//	of the direction vector, and I might have picked integers
+	//	and thus the test might not have "passed"
+	//	---
+	//	so i calculated the number of this by hand.
+	//	then i wrote a test expecting it to pass
+	//	it passed with "* 2"
+	//	it passed with " << 1" because the direction vector was integers
+	const k1Times2xxx = k1 << 1;
```

Comment:
Just leaving this here for posterity.
Yup, bit shifting works great for integers.
No so much when the underlying number is a decimal/fraction/double, whatever.